### PR TITLE
upgrade finagle

### PR DESCRIPTION
**Pull request by crispywalrus**
to 17.11.0 and make a few changes to support. Added finagle-netty3
dependency and updated the uses of various ssl related usages.

Issues:

+ ~~the docs don't build. I get a "key not found" for PG_USER~~
+ ~~the SslClientSessionVerifier is fairly different and I'm not sure I made the right changes~~
+ ~~I chose (based on issues with other twitter/finagle projects) to change the version to make the rest of the twitter ecosystem. This is clearly correct unless it's not.~~

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -3,16 +3,18 @@ import com.typesafe.sbt.SbtGhPages.GhPagesKeys._
 
 lazy val buildSettings = Seq(
   organization := "io.github.finagle",
-  version := "0.4.3",
-  scalaVersion := "2.12.2",
-  crossScalaVersions := Seq("2.11.11","2.12.2")
+  version := "17.11.0",
```
Comment (jeremyrsmith) [inline build.sbt]: Let's make this `0.5.0` - I don't think we have to follow the same version numbering of Finagle. Major version 17 implies a level of stability that I'm not sure I'd assert for finagle-postgres.

Other than that, LGTM!

File: src/test/scala/com/twitter/finagle/postgres/integration/IntegrationSpec.scala [hunk 2]
```diff
@@ -411,15 +411,15 @@ class IntegrationSpec extends Spec {
       }
 
       "return correct availability information" when {
-        "client is good" in {
+        "client is good" ignore {
```
Comment (jeremyrsmith) [inline src/test/scala/com/twitter/finagle/postgres/integration/IntegrationSpec.scala]: How come these are ignored? They don't pass?

Comment (crispywalrus) [reply src/test/scala/com/twitter/finagle/postgres/integration/IntegrationSpec.scala]: Yes, the state management inside finagle has changed even for netty3 services

Comment (jeremyrsmith) [reply src/test/scala/com/twitter/finagle/postgres/integration/IntegrationSpec.scala]: If these APIs no longer work then I feel like we should either remove them (to lampshade the breaking change) or make them work. 

@clhodapp added this functionality, maybe he can take a look at the issue or comment whether it's still relevant in the newer finagle structure?

Comment (clhodapp) [reply src/test/scala/com/twitter/finagle/postgres/integration/IntegrationSpec.scala]: @crispywalrus 

I get reliable passes on a laptop if I just update the middle test to:
```
        "client is bad" in {
          val badClient: PostgresClient = getBadClient
          badClient.isAvailable must equal(false)
          Set(Status.Busy, Status.Closed) must contain (badClient.status)
        }
```

Comment (leonmaia) [reply src/test/scala/com/twitter/finagle/postgres/integration/IntegrationSpec.scala]: Same for me, got this test passing all the time. With basically the same code as @clhodapp but I took the inverse approach:
```
"client is bad" in {
          val badClient: PostgresClient = getBadClient
          badClient.isAvailable must equal(false)
          badClient.status mustNot equal(Status.Open)
        }
```

**Comment by jeremyrsmith** [comment]
Thanks, LGTM @crispywalrus

**Comment by crispywalrus** [comment]
definitely like to call out @clhodapp for the help on the tests, thanks chris

**Comment by clhodapp** [comment]
Happy to help, Chris 😸

**Pull request merged**